packages/@uppy/transloadit/src/Assembly.ts

```diff
@@ -101,16 +101,30 @@ class TransloaditAssembly extends Emitter {
     })
 
     this.#sse.addEventListener('assembly_upload_finished', (e) => {
-      const file = JSON.parse(e.data)
-      this.status.uploads.push(file)
+      const file = JSON.parse(e.data) as AssemblyFile
+      let uploads = this.status.uploads
+      if (!uploads) {
+        uploads = [] as NonNullable<AssemblyResponse['uploads']>
+        this.status.uploads = uploads
+      }
+      uploads.push(file)
       this.emit('upload', file)
     })
 
     this.#sse.addEventListener('assembly_result_finished', (e) => {
-      const [stepName, result] = JSON.parse(e.data)
+      const [stepName, rawResult] = JSON.parse(e.data) as [
+        string,
+        AssemblyResult,
+      ]
+      rawResult.localId ??= null
+      let results = this.status.results
+      if (!results) {
+        results = {} as NonNullable<AssemblyResponse['results']>
+        this.status.results = results
+      }
```

Comment:
Same here, this wasn't needed at all before, why now?
```suggestion
      this.status.results ??= {}
```